ludwig/api.py

```diff
@@ -891,6 +910,46 @@ def _tune_batch_size(self, trainer, dataset, random_seed: int = default_random_s
         trainer.eval_batch_size = self.config_obj.trainer.eval_batch_size
         trainer.gradient_accumulation_steps = self.config_obj.trainer.gradient_accumulation_steps
 
+    def generate(
+        self,
+        input_strings: Union[str, List[str]],
+        generation_config: Optional[dict] = None,
+    ) -> Union[str, List[str]]:
+        """A simple generate() method that directly uses the underlying transformers library to generate text."""
+        if self.config_obj.model_type != MODEL_LLM:
+            raise ValueError(
+                f"Model type {self.config_obj.model_type} is not supported by this method. Only `llm` model type is "
+                "supported."
+            )
+        if not torch.cuda.is_available():
+            raise ValueError("GPU is not available.")
+
+        if not self.model.model.config.is_encoder_decoder:
+            padding_side = "left"
+        else:
+            padding_side = "right"
+        tokenizer = HFTokenizer(self.config_obj.base_model, padding_side=padding_side)
```

Comment:
Do we want to use the tokenizer initialized in `self.model.tokenizer` once https://github.com/ludwig-ai/ludwig/pull/3745 lands?

Reply:
There's extra handling for the padding side that I had to add in this function to bypass warnings from HF.

I've added a TODO to figure out if this should also be folded into `llm.py`.

Regardless, we should still consolidate the tokenizer initialization in `llm.py`.